docs/internal/types.md

```diff
@@ -473,3 +477,23 @@ class MyClass : EqualComparable, Addable
 ```
 
 In the example above, the `MyClass` argument type _matches_ the `Self` type in the interface declaration because it is the same type as the class declaring it.
+
+If instead we had to use the interface's name, the methods `==` and `+` would match _any_ types that implement the interfaces.
+
+Example:
+```ts
+interface Addable
+{
+  +(Addable & readonly, Addable & readonly) : Addable & imm;
+}
+class Foo : Addable { ... }
+class Bar : Addable { ... }
+class Baz : Addable { ... }
+
+function() {
+  foo = Foo;
+  bar = Bar;
+  // This matches the interface, but not Foo's (or Bar's) implementation
+  baz : Baz & imm = Addable::+(foo, bar);
```

Comment:
This seems to be calling a method that doesn't have an implementation.

Reply:
I'm still a bit fuzzy on the syntax for calls through interfaces.

I didn't want to call `foo.+(bar)` or `foo + bar` because that has a lot of context that is irrelevant to the example.

Is there a clear way to show that this is the call that the interface mandates? Calling `Foo::+(foo, bar)` isn't really the same thing.

Perhaps I should just show two concrete classes, where one implements `+(One, One)` (which works) and the other `+(One, Two)` which also works but isn't what you meant.

Reply:
Ok, so now I've gone some implementations and the final example should be clearer.